antsibull/schemas/docs/base.py

```diff
@@ -425,6 +425,18 @@ class SeeAlsoLinkSchema(BaseModel):
     name: str
 
 
+class AttributeSchema(BaseModel):
+    description: str
+    support: str = p.Field('str', regex='^(full|partial|none)$')
```

Comment:
Is support=='none' supposed to mean, "this plugin does not support this feature"?  If so, maybe we want the schema to generate default values for the attributes where support is set to none.

That way, conditionals that look for what features are supported can do `if plugin.attributes.action.support != 'none'` rather than `if 'action' in plugin.attributes and plugin.attributes.action.support != 'none'`.

If support none means something else, then we can't do that, of course.

Reply:
Yes, `support=='none'` means the plugin doesn't support this feature.

I'm not really sure how these things should actually be used in the docs, though, since only very few plugins actually extend the common attributes doc fragment (https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/doc_fragments/action_common_attributes.py) (in fact, it will take a long time until a plugin outside ansible-core will do so to not break compatibility with ansible-core < 2.12), and plugins in general might define their own attributes (which could use the default names, but with a different meaning).

@bcoca you might want to weight in here.